example_12/controllers/include/passthrough_controller/passthrough_controller.hpp

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) 2023, PAL Robotics
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PASSTHROUGH_CONTROLLER__PASSTHROUGH_CONTROLLER_HPP_
+#define PASSTHROUGH_CONTROLLER__PASSTHROUGH_CONTROLLER_HPP_
+
+// system
+#include <limits>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <controller_interface/chainable_controller_interface.hpp>
+#include "controller_interface/helpers.hpp"
+#include "realtime_tools/realtime_buffer.h"
+#include "std_msgs/msg/float64_multi_array.hpp"
+// auto-generated by generate_parameter_library
+#include "passthrough_controller_parameters.hpp"
+
+#include "passthrough_controller/visibility_control.h"
+
+namespace passthrough_controller
+{
+using DataType = std_msgs::msg::Float64MultiArray;
+class PassthroughController : public controller_interface::ChainableControllerInterface
+{
+public:
+  PASSTHROUGH_CONTROLLER__VISIBILITY_PUBLIC
+  PassthroughController() {}
+
+  PASSTHROUGH_CONTROLLER__VISIBILITY_PUBLIC
+  controller_interface::CallbackReturn on_init() override
+  {
+    try
+    {
+      param_listener_ = std::make_shared<ParamListener>(get_node());
+      params_ = param_listener_->get_params();
+    }
+    catch (const std::exception & e)
+    {
+      fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
+      return controller_interface::CallbackReturn::ERROR;
+    }
+    return controller_interface::CallbackReturn::SUCCESS;
+  }
+
+  PASSTHROUGH_CONTROLLER__VISIBILITY_PUBLIC
+  controller_interface::CallbackReturn on_configure(const rclcpp_lifecycle::State &) override
+  {
+    params_ = param_listener_->get_params();
+    command_interface_names_ = params_.interfaces;
+
+    if (command_interface_names_.empty())
+    {
+      RCLCPP_ERROR(get_node()->get_logger(), "'interfaces' parameter list was empty");
+      return controller_interface::CallbackReturn::ERROR;
+    }
```

Comment:
Shouldn't we use the validator `not_empty<> `of generate parameter lib instead?

Reply:
You are right. I've modified it to use the validators.